app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileView.kt

```diff
@@ -0,0 +1,244 @@
+              .background(Color.LightGray)
+              .testTag("profileHeader")) {
+        // Profile image
+        Image(
+            painter = painterResource(id = R.drawable.turtle),
+            contentDescription = "Profile Image",
+            contentScale = ContentScale.Crop,
+            modifier = Modifier.fillMaxSize().testTag("profilePicture"))
+
+        // Back button
+        IconButton(
+            onClick = { navigationActions.goBack() },
+            modifier =
+                Modifier.align(Alignment.TopStart) // Align to the top start (top-left corner)
+                    .padding(16.dp)
+                    .testTag("goBackButton")) {
+              Icon(
+                  imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                  contentDescription = "Go Back",
+                  tint = Color.White)
+            }
+
+        // Overlay with username and edit button
+        Row(
+            modifier =
+                Modifier.align(Alignment.BottomStart) // Aligns this content to the bottom left
+                    .fillMaxWidth()
+                    .padding(16.dp),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween) {
+
+              // Username
+              Text(
+                  text = "Jean Michelle",
+                  fontSize = 24.sp,
+                  fontWeight = FontWeight.Bold,
+                  color = Color.White,
+                  modifier = Modifier.testTag("username"))
+
+              // Edit Button
+              IconButton(
+                  onClick = { navigationActions.navigateTo("Profile Edit Screen") },
+                  modifier =
+                      Modifier.background(Color.White, CircleShape)
+                          .padding(4.dp)
+                          .testTag("editButton")) {
+                    Icon(
+                        imageVector = Icons.Filled.Create,
+                        contentDescription = "Edit Profile",
+                        tint = Color.Gray)
+                  }
+            }
+      }
+}
+
+/**
+ * Displays the user's catchphrase in a medium-sized font with a maximum of two lines.
+ *
+ * @param catchPhrase The user's catchphrase.
+ */
+@Composable
+fun ProfileCatchPhrase(catchPhrase: String) {
+  Text(
+      text = catchPhrase,
+      style = MaterialTheme.typography.bodyMedium,
+      color = Color.Black.copy(alpha = 0.8f),
+      fontWeight = FontWeight.Medium,
+      fontSize = 16.sp,
+      textAlign = TextAlign.Left,
+      maxLines = 2,
+      overflow = TextOverflow.Ellipsis)
+}
+
+/**
+ * Displays a section with a row of tags, each associated with a color.
+ *
+ * @param listOfTags A list of tags, each paired with a color.
+ */
+@Composable
+fun TagsSection(listOfTags: List<Pair<String, Color>>) {
+  Box(modifier = Modifier.fillMaxWidth().height(80.dp).testTag("tagSection")) {
```

Comment:
Don't use magic values, define them as constants for better maintainability: 

private const val TAG_SECTION_HEIGHT = 80.dp